ArduCopter/GCS_Mavlink.cpp

```diff
@@ -139,20 +140,26 @@ void GCS_MAVLINK_Copter::send_position_target_local_ned()
         type_mask = POSITION_TARGET_TYPEMASK_VX_IGNORE | POSITION_TARGET_TYPEMASK_VY_IGNORE | POSITION_TARGET_TYPEMASK_VZ_IGNORE |
                     POSITION_TARGET_TYPEMASK_AX_IGNORE | POSITION_TARGET_TYPEMASK_AY_IGNORE | POSITION_TARGET_TYPEMASK_AZ_IGNORE |
                     POSITION_TARGET_TYPEMASK_FORCE_SET | POSITION_TARGET_TYPEMASK_YAW_IGNORE| POSITION_TARGET_TYPEMASK_YAW_RATE_IGNORE; // ignore everything except position
-        target_pos = copter.wp_nav->get_wp_destination() * 0.01f; // convert to metres
+        target_pos = copter.mode_guided.get_target_pos() * 0.01f; // convert to metres
         break;
-    case ModeGuided::SubMode::Velocity:
+    case ModeGuided::SubMode::Accel:
```

Comment:
how about we keep the order here the same as the enum?  If that means changing the order of the enum so that "Accel" is before "VelAccel" then that is OK i think.  the order of the enum will only show up to users in the form of the logging.

Reply:
enum class SubMode {
        TakeOff,
        WP,
        PosVelAccel,
        VelAccel,
        Accel,
        Angle,
    };
    
    take off then max autonomy to minimum autonomy.